jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpTransportOverHTTP2.java

```diff
@@ -396,9 +397,10 @@ public void failed(Throwable failure)
             synchronized (this)
             {
                 commit = this.commit;
-                // Only fail pending writes, as we
-                // may need to write an error page.
-                if (state == State.WRITING)
+                // Don't always fail, as we may need to write an error response.
+                EndPoint endPoint = connection.getEndPoint();
+                boolean cannotWrite = stream.isReset() || endPoint.isOutputShutdown() || !endPoint.isOpen();
+                if ((cannotWrite && state == State.IDLE) || state == State.WRITING)
```

Comment:
@gregw suggests that we check first for the `callback` and if there, we fail it (to be sure that we fail it if there is one).
Then we can move to FAILED state if there is no callback but we cannot write.

Reply:
Or just go to the FAILED state and be done with it.